### PR TITLE
Add ratios endpoint to proxy calls to Coingecko coin markets endpoint

### PR DESCRIPTION
**Pull request by nvonpentz**
### Summary
Resolves https://github.com/brave-intl/bat-go/issues/1225

Add ratios endpoint to proxy calls to Coingecko coin markets endpoint

Client calls to new ratios endpoint `/v2/market/provider/coingecko` will be proxied to`/api/v3/coins/markets` on [coingecko](https://www.coingecko.com/en/api/documentation).  The ratios endpoint currently supports two query string parameters `limit` which specifies the number of results (max 250), and `vsCurrency` which specifies the currency the results are denominated in.

The results for a given call are saved on Redis for one hour.  During that time, calls to the same endpoint will be routed to the Redis cache, not coingecko.

Both the path and query supported query strings are flexible, so let me know if something else is desired.

```
$ curl {host}/v2/market/provider/coingecko\?vsCurrency\=usd\&limit\=2 | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   641  100   641    0     0    768      0 --:--:-- --:--:-- --:--:--   767
{
  "payload": [
    {
      "id": "bitcoin",
      "symbol": "btc",
      "name": "Bitcoin",
      "image": "https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579",
      "market_cap": 836181040930,
      "market_cap_rank": 1,
      "current_price": 44073,
      "price_change_24h": 80.62,
      "price_change_percentage_24h": 0.18327,
      "total_volume": 25839488245
    },
    {
      "id": "ethereum",
      "symbol": "eth",
      "name": "Ethereum",
      "image": "https://assets.coingecko.com/coins/images/279/large/ethereum.png?1595348880",
      "market_cap": 355144090752,
      "market_cap_rank": 2,
      "current_price": 0,
      "price_change_24h": 12.74,
      "price_change_percentage_24h": 0.43069,
      "total_volume": 17259869597
    }
  ],
  "lastUpdated": "2022-03-02T20:35:53.40707037Z"
}
```

### Type of change ( select one )

- [ ] Product feature
- [ ] Bug fix
- [ ] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before submitting this PR:

- [x] Does your code build cleanly without any errors or warnings?
- [x] Have you used auto closing keywords?
- [x] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [ ] Is there a clear title that explains what the PR does?
- [x] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security / privacy review if needed
- [ ] Have you performed a self review of this PR?

### CODE REVIEW

File: utils/clients/coingecko/client.go [hunk 1]
```diff
@@ -289,3 +290,114 @@ func (c *HTTPClient) FetchSupportedVsCurrencies(ctx context.Context) (*Supported
...
+) (*CoinMarketResponse, time.Time, error) {
+	updated := time.Now()
+	url := "/api/v3/coins/markets"
+	params := &coinMarketParams{
+		baseParams: c.baseParams,
+		VsCurrency: vsCurrency,
+		Limit:      limit,
+	}
+
+	cacheKey, err := c.cacheKey(ctx, url, params)
```
Comment (husobee) [inline utils/clients/coingecko/client.go:341]: Would it be better to just always get from CG the top 250 and cache that?  As I read this, we are using the limit in the cache key.

Suggest always getting top 250 from CG and then in only return the `limit` number of responses from what has been cached.

**Review by husobee — CHANGES_REQUESTED** [review]
Only concern is the caching including the user defined limit.

File: utils/clients/coingecko/client.go (hunk 1 above)
Comment (nvonpentz) [reply utils/clients/coingecko/client.go:341]: Yes, that's better, will adjust the logic.  Thanks

Comment (nvonpentz) [reply utils/clients/coingecko/client.go:350]: 2f2d1e38f0d9b050683711f1e2b0e94b62011d55

**Review by husobee — APPROVED** [review]
lgtm

**Pull request merged**